### PR TITLE
Show price in alternative currency specified

### LINKED ISSUE #546: Display fiat equivalents next to BTC amounts
Any time a BTC amount is displayed (Total balance, confirmed+pending; amount in History tab; amount in UTXO tab; amount in Send New + Unsigned tab; etc, etc), it would be neat to have the fiat value right after, eg: 

"0.1 ($130)"

The exchange rate could be a setting on the Settings general tab, much like block explorer is. 

Top right corner of the screen where Tor icon + Settings icon is, perhaps could display full exchange rate there as well, eg ($13000). 

Additionally, it'd be neat to allow the user to manually set a exchange rate value. eg., if click the Fiat-value (or placeholder "Set fiat value") button just (eg) left hand side of Tor icon, the user can manually enter a value. This'd remove the reliance on third party source as an optionality to the user. For setting value manually, could have a currency ticker dropdown too, so user's could both manually enter (exchangeRate) + (currencyTicker) to display in their currency of choice. 

Below some quick examples of how it might look in some of the places (excuse the incorrect proportions on the Set value/exchange rate left hand side of Tor icon; done via browser devtools)

SpecterD, 1
SpecterD
SpecterD, 3
SpecterD,4

### PR DESCRIPTION
**Pull request by ben-kaufman**
Close #546 

Allow user to specify (in the general settings page) a currency symbol and price of BTC and display that next to amounts we show on Specter.

### CODE REVIEW

**Comment by stepansnigirev** [comment]
Maybe we could have a selector - either to enter the price manually or select the source where we should get the price?

For example we could support:
- coindesk with urls like https://api.coindesk.com/v1/bpi/currentprice/EUR.json (supports many currencies)
- bitstamp like https://www.bitstamp.net/api/v2/ticker/btcusd
- more sources depending on what people want

**Comment by ChristopherA** [comment]
You might want to consider supporting spotbit.info via https or tor

**Comment by stepansnigirev** [comment]
> You might want to consider supporting spotbit.info via https or tor

Trying it out, http://h6zwwkcivy2hjys6xpinlnz2f74dsmvltzsd4xb42vinhlcaoe7fdeqd.onion/now/USD 
getting "Internal Server Error"

File: src/cryptoadvance/specter/controller.py [hunk 1]
```diff
@@ -1786,6 +1794,18 @@ def btcunitamount(value):
     return "{:,.0f}".format(round(value * 1e8))
 
 
+@app.template_filter("altunit")
+def altunit(value):
+    if app.specter.alt_rate and app.specter.alt_symbol:
+        return (
+            "{:,.2f}".format(float(value) * float(app.specter.alt_rate))
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/controller.py:1801]: Add space here:
```suggestion
            "{:,.2f} ".format(float(value) * float(app.specter.alt_rate))
```

File: src/cryptoadvance/specter/templates/settings/general_settings.jinja [hunk 2]
```diff
@@ -53,6 +53,12 @@
 				<option value="sat" {% if unit=="sat" %} selected="selected"{% endif %}>sats</option>
 			</select>
 			<br><br>
+			<span>Alternative currency rate (per BTC):</span>
+			<input type="number" min="0" step="0.01" name="alt_rate" value="{{ alt_rate }}" placeholder="Set the price of bitcoin in your chosen currency">
+			<br><br>
+			Alternative currency symbol:
+			<input type="text" name="alt_symbol" value="{{ alt_symbol }}" placeholder="eg. $/€/£">
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/templates/settings/general_settings.jinja:60]: Maybe put them in a row?
```suggestion
			<div class="row">
				<input type="number" min="0" step="0.01" name="alt_rate" value="{{ alt_rate }}" placeholder="Set the price of bitcoin in your chosen currency" style="flex-grow: 1">
				&nbsp;
				<input type="text" name="alt_symbol" value="{{ alt_symbol }}" placeholder="eg. $/€/£" style="max-width: 30px">
			</div>
```

File: src/cryptoadvance/specter/controller.py [hunk 3]
```diff
@@ -393,10 +393,14 @@ def general_settings():
     explorer = app.specter.explorer
     loglevel = get_loglevel(app)
     unit = app.specter.unit
+    alt_rate = app.specter.alt_rate
+    alt_symbol = app.specter.alt_symbol
     if request.method == "POST":
         action = request.form["action"]
         explorer = request.form["explorer"]
         unit = request.form["unit"]
+        alt_rate = request.form.get("alt_rate", 0)
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/controller.py:402]:
```suggestion
        alt_rate = float(request.form.get("alt_rate", 0))
```

File: src/cryptoadvance/specter/specter.py [hunk 4]
```diff
@@ -477,6 +479,20 @@ def update_unit(self, unit, user):
         else:
             user.set_unit(self, unit)
 
+    def update_alt_rate(self, alt_rate, user):
+        if user.is_admin:
+            self.config["alt_rate"] = alt_rate
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/specter.py:484]:
```suggestion
            self.config["alt_rate"] = float(alt_rate)
```

File: src/cryptoadvance/specter/user.py [hunk 5]
```diff
@@ -108,6 +108,14 @@ def set_unit(self, specter, unit):
         self.config["unit"] = unit
         self.save_info(specter)
 
+    def set_alt_rate(self, specter, alt_rate):
+        self.config["alt_rate"] = alt_rate
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/user.py:112]:
```suggestion
        self.config["alt_rate"] = float(alt_rate)
```

**Comment by s-5Q2X8mgeGR** [comment]
If you're going the route of allowing user to manually enter the BTCFIAT value, the user may want to update this amount often. Would suggest (also perhaps) displaying it to the left hand side of Tor icon at top right corner, thus always visible and easily changed, no need to go to general settings page. (This shown in the quick mockup on https://github.com/cryptoadvance/specter-desktop/issues/546)

**Comment by pxsocs** [comment]
Including this as it may be helpful. [This library](https://github.com/pxsocs/specter_warden/blob/master/src/warden/warden_pricing_engine.py) has the ability to use different sources for price feed in USD and any other currencies. The `PriceData` class is the relevant on here. 

Sample usage:
```python
btc = PriceData("BTC", "cc_digital")
```
Then:
btc.errors:         Any error messages
btc.provider:     Provider being used for requests
btc.filename:     Local filename where historical prices are saved

Other info:
btc.ticker, btc.last_update, btc.first_update, btc.last_close
btc.update_history(force=False)
btc.df_fx(currency, fx_provider): returns a df with
                                  prices and fx conversions
btc.price_ondate(date)
btc.price_parser(): do not use directly. This is used to parse
                    the requested data from the API provider
btc.realtime(provider): returns realtime price (float)

**Comment by javiervargas** [comment]
> > You might want to consider supporting spotbit.info via https or tor
> 
> 
> 
> Trying it out, http://h6zwwkcivy2hjys6xpinlnz2f74dsmvltzsd4xb42vinhlcaoe7fdeqd.onion/now/USD 
> 
> getting "Internal Server Error"

Please try this:  http://h6zwwkcivy2hjys6xpinlnz2f74dsmvltzsd4xb42vinhlcaoe7fdeqd.onion/now/usd/coinbase    For other exchanges look at https://spotbit.info.

**Comment by ben-kaufman** [comment]
Replaced with #622

**Pull request closed without merging**